File: Examples/ClassDiagram/Java/Podcast.java
package medialibrary;

/** One episode of a podcast. */
public final class Podcast extends MediaItem {
    private final String host;
    private final int episodeNumber;

    public Podcast(String title, double duration, String host, int episodeNumber) {
        super(title, duration, Genre.SPOKEN_WORD);
        this.host = host;
        this.episodeNumber = episodeNumber;
    }

    @Override
    public void play() {
        System.out.println("🎙 Episode " + episodeNumber + ": " + title());
    }
}

File: Examples/ClassDiagram/Kotlin/Podcast.kt
package medialibrary

/** One episode of a podcast. */
class Podcast(
    title: String,
    duration: Double,
    val host: String,
    val episodeNumber: Int
) : MediaItem(title, duration, Genre.SPOKEN_WORD) {
    override fun play() {
        println("🎙 Episode $episodeNumber: $title")
    }
}
